src/theme/ods-chart-theme.ts

```diff
@@ -403,6 +403,10 @@ export class ODSChartsTheme {
         'the chart basic options must be set to get the theme completion'
       );
     }
+    // need to copy dataOptions as theme feature may change the original dataOptions
+    // only make an asign at the first level in order to avoid deep copy of all data
+    // each feature is responsible to deep copy the changed part according to their changes
+    this.dataOptions = { ...this.dataOptions };
```

Comment:
I want to avoid copying all the options in depth because they include the data to be displayed, which can include several thousand points.
So I only copy the top-level references, the referenced sub-objects remain the same. 
But this allows a feature to then change this reference by copying the top-level attribute in depth, and only this top-level attribute. 
For example, the instructions :
`this.dataOptions = { ...this.dataOptions };`
then
`dataOptions.tooltip = cloneDeepObject(dataOptions.tooltip);`
do that the initial object is unchanged, it is not entirely recopied while still allowing modifications to be made to dataOptions.tooltip in our code.

Reply:
Ohh, ok I think I get it, thanks for the explanation 🙏 I'm fine with this then